src/main/java/org/icgc/argo/program_service/grpc/ProgramServiceImpl.java

```diff
@@ -224,6 +235,26 @@ public void listUser(ListUserRequest request, StreamObserver<ListUserResponse> r
     responseObserver.onCompleted();
   }
 
+  private Invitation getInvitationForEgoUser(String programShortName, User user) {
```

Comment:
Can you move `getInvitationForEgoUser` to `InvitationService`? It's better to keep all the service logic within service layer.